gobblin-metrics-libs/gobblin-metrics-base/src/main/java/gobblin/metrics/broker/MetricContextFactory.java

```diff
@@ -68,7 +82,12 @@ public String getName() {
         parentMetricContext = broker.getSharedResourceAtScope(this, config.getKey(), parentScope);
       }
 
-      MetricContext.Builder builder = parentMetricContext.childBuilder(broker.selfScope().toString());
+      // If this is the root scope, append a UUID to the name. This allows having a separate root context per broker.
+      String metricContextName = parents == null ?
+          config.getScope().name() + "_" + UUID.randomUUID().toString() :
```

Comment:
Does that mean each time we call createResource against root scope broker, we will have different metricContextName generated? What is the purpose of this?